internal/client/githubclient/githubclient.go

```diff
@@ -0,0 +1,161 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package githubclient
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/google/go-github/github"
+	"github.com/guacsec/guac/internal/client"
+	"golang.org/x/oauth2"
+)
+
+// Wrapper interface for Github
+type GithubClient interface {
```

Comment:
Yep. I was going to wait until we needed to for gitlab or similar.

Reply:
yea this is good but at the same time, i'd be good to just slap a TODO on since if gitlab functionality gets implemented, moving this out will probably be part of that whole refactor, as of now, its simpler to understand.

Reply:
Sure, I have the TODO in the repo_client where it would most likely live but yeah probably makes sense to have one here too